src/components/network-map-viewer/network/network-map.jsx

```diff
@@ -523,6 +528,118 @@ const NetworkMap = (props) => {
         setCentered(INITIAL_CENTERED);
     }, [mapLib?.key]);
 
+    const [features, setFeatures] = useState({});
+
+    useEffect(() => {
+        props.onFeaturesChanged(features);
+    }, [features]);
+
+    const onUpdate = useCallback((e) => {
+        setFeatures((currFeatures) => {
+            const newFeatures = { ...currFeatures };
+            for (const f of e.features) {
+                newFeatures[f.id] = f;
+            }
+            return newFeatures;
+        });
+    }, []);
+
+    const getPolygonFeatures = () => features;
+    const getSelectedSubstations = () => {
+        const substations = getSubstationsInPolygon(
+            features,
+            props.mapEquipments,
+            props.geoData
+        );
+        return (
+            substations.filter((substation) => {
+                return substation.voltageLevels.some((vl) =>
+                    props.filteredNominalVoltages.includes(vl.nominalV)
+                );
+            }) ?? []
+        );
+    };
+
+    const getSelectedVoltageLevel = () => {
+        const selectedVL = getVoltageLevelFromSubstation(
+            getSelectedSubstations()
+        );
+        return selectedVL.filter((vl) =>
+            props.filteredNominalVoltages.includes(vl.nominalV)
+        );
+    };
+
+    function getSelectedLinesInPolygon(
+        network,
+        lines,
+        geoData,
+        polygonCoordinates
+    ) {
+        return lines.filter((line) => {
+            try {
+                const linePos = geoData.getLinePositions(network, line);
+                if (!linePos) {
+                    return false;
+                }
+
+                return (
+                    booleanPointInPolygon(linePos[0], polygonCoordinates) &&
+                    booleanPointInPolygon(linePos[1], polygonCoordinates)
```

Comment:
If one of the linePos is insite the polygon we should keep the line I think.
PO seems OK with this rule.
```suggestion
                    booleanPointInPolygon(linePos[0], polygonCoordinates) ||
                    booleanPointInPolygon(linePos[1], polygonCoordinates)
```

Reply:
We need even more, because in detailed mode we could have a lot more than 2 pos
```diff
return (
-    booleanPointInPolygon(linePos[0], polygonCoordinates) &&
-    booleanPointInPolygon(linePos[1], polygonCoordinates)
+    linePos.some((pos) => booleanPointInPolygon(pos, polygonCoordinates)) // at least one point in polygon then OK
);
```